### DISCUSSION TITLE
Simulation crash when long press to insert a nearby POI into the route

### DISCUSSION THREAD

**Discussion by DolomitesTom** [devemux86/cruiser#263] [Q&A] [answered]
First I need to say that this function is great when it works. It should be refined: at the moment also nearby POIs which are in the back, are also shown as nPOI. This should be changed (if possible) that only such POIs and favorites are shown which are ahead, not already passed. Only POIs which are located from actual GPS position in driving direction should be shown.

@devemux86 
have sent you also my prefs and example by email. 

I have activated Germany, France and Luxemburg maps and POI, nothing else.

The crashes happen on both phones, S8+ (Android 10) and A52S (Android 13).

At the moment I only use simulation, 4x speed, it crashes also on 1x speed.
![grafik](https://github.com/devemux86/cruiser/assets/121371629/7af0c07f-6f45-40f6-a974-59dfb9d814b9)

I hope you can simulate it too.

**Comment by devemux86** [comment]
@DolomitesTom thanks for the report.

> At the moment I only use simulation, 4x speed, it crashes also on 1x speed.

It only happens in the simulation that works with special rules, the _real time navigation works correctly_.

Long press will also work properly in simulation in the next Cruiser update.

**Comment by DolomitesTom** [comment]
perfect, thanks, is this somehow possible:
At the moment also nearby POIs which are in the back, are also shown as nPOI. This should be changed (if possible) that only such POIs and favorites are shown which are ahead, not already passed. Only POIs which are located from actual GPS position in driving direction should be shown.
or is this also only an issue in simulation because there is no GPS active?

**Reply by devemux86** [reply]
> or is this also only an issue in simulation because there is no GPS active?

Simulation uses a virtual GPS location.

Only favorites and POI that are ahead of user location on the route are displayed / announced.

Perhaps it was a side effect of the previous issue in simulation.
How can you verify the "in the back"? Can you record a video?

I cannot reproduce it, although now I test with the fixed version.

**Reply by DolomitesTom** [reply]
> Only favorites and POI that are ahead of user location on the route are displayed / announced.

perfect

> Perhaps it was a side effect of the previous issue in simulation.

possibly

> How can you verify the "in the back"? 
I paused the simulation and checked where the wp was created. It was behind the simulated actual location.

> Can you record a video?

no sorry

**Reply by devemux86** [reply]
> I paused the simulation and checked where the wp was created. It was behind the simulated actual location.

Simulation works a bit differently, it starts on the nearest turn instruction.

However, it is not expected to change the route while simulation runs,
as with long press nearby markers, so perhaps we should not allow it.

**Reply by DolomitesTom** [reply]
> However, it is not expected to change the route while simulation runs

Maybe, but without it, I couldn't have pushed Cruiser to the limits 😁.

**Comment by devemux86** [comment, marked as answer]
Fixed in [Cruiser 3.0.29](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
